<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1088</width>
    <height>705</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QVBoxLayout" name="verticalLayout" stretch="1,14,0,10">
      <item>
       <layout class="QHBoxLayout" name="topNavBar" stretch="0,0,0,0,0,0,0,0">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Arduino Baud Rate:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="BaudRate_selector">
          <item>
           <property name="text">
            <string>300</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>600</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>1200</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>2400</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>4800</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>9600</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>14400</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>19200</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>28800</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>38400</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>57600</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>115200</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="COMPort_selector"/>
        </item>
        <item>
         <widget class="QPushButton" name="arduino_connect">
          <property name="text">
           <string>Connect</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>RTSP URL:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="rtsp_input">
          <property name="text">
           <string>rtsp://192.168.0.9:8554/live.sdp</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="rtsp_connect">
          <property name="text">
           <string>Connect</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="1,1">
        <item>
         <widget class="MpvWidget" name="VideoWidget"/>
        </item>
        <item>
         <widget class="AvgImgPlot" name="liveLinePlot" native="true"/>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QTabWidget" name="tabWidget">
        <property name="tabPosition">
         <enum>QTabWidget::North</enum>
        </property>
        <property name="currentIndex">
         <number>1</number>
        </property>
        <widget class="QWidget" name="Arduino">
         <attribute name="title">
          <string>Arduino</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="3,5,7">
            <item>
             <widget class="QGroupBox" name="groupBox">
              <property name="enabled">
               <bool>true</bool>
              </property>
              <property name="title">
               <string>Arduino Controls</string>
              </property>
              <property name="flat">
               <bool>false</bool>
              </property>
              <layout class="QHBoxLayout" name="horizontalLayout_3">
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_5">
                 <item>
                  <layout class="QVBoxLayout" name="verticalLayout_3">
                   <item>
                    <widget class="QPushButton" name="Arduino_Start">
                     <property name="text">
                      <string>Start</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QPushButton" name="Arduino_Stop">
                     <property name="text">
                      <string>Stop</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QPushButton" name="Arduino_Push">
                     <property name="text">
                      <string>Push</string>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QGroupBox" name="groupBox_2">
              <property name="maximumSize">
               <size>
                <width>160000</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="title">
               <string>Graph controls</string>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_4" stretch="0,0,1,0">
               <item>
                <layout class="QHBoxLayout" name="layout123" stretch="2,5">
                 <property name="topMargin">
                  <number>0</number>
                 </property>
                 <item>
                  <widget class="QLabel" name="Px_Pr_CM_Label">
                   <property name="text">
                    <string>Px. pr. cm:</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="Pixels_Per_CM_Input">
                   <property name="minimumSize">
                    <size>
                     <width>0</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="maximumSize">
                    <size>
                     <width>16777215</width>
                     <height>16777215</height>
                    </size>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_7" stretch="2,5">
                 <property name="topMargin">
                  <number>0</number>
                 </property>
                 <item>
                  <widget class="QLabel" name="Dist_To_Pusher_Label">
                   <property name="minimumSize">
                    <size>
                     <width>0</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="text">
                    <string>Dist. to Pusher: </string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="Dist_To_Pusher_Input">
                   <property name="minimumSize">
                    <size>
                     <width>0</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="maximumSize">
                    <size>
                     <width>16777215</width>
                     <height>16777215</height>
                    </size>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout" stretch="4,2,8">
                 <property name="spacing">
                  <number>6</number>
                 </property>
                 <item>
                  <widget class="QLabel" name="label_3">
                   <property name="text">
                    <string>Avg. trigger:</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLineEdit" name="refLineInput">
                   <property name="minimumSize">
                    <size>
                     <width>0</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="maximumSize">
                    <size>
                     <width>50</width>
                     <height>16777215</height>
                    </size>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QSlider" name="refLineSlider">
                   <property name="minimumSize">
                    <size>
                     <width>0</width>
                     <height>0</height>
                    </size>
                   </property>
                   <property name="maximumSize">
                    <size>
                     <width>16777215</width>
                     <height>16777215</height>
                    </size>
                   </property>
                   <property name="autoFillBackground">
                    <bool>false</bool>
                   </property>
                   <property name="inputMethodHints">
                    <set>Qt::ImhNone</set>
                   </property>
                   <property name="maximum">
                    <number>255</number>
                   </property>
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="invertedAppearance">
                    <bool>false</bool>
                   </property>
                   <property name="invertedControls">
                    <bool>false</bool>
                   </property>
                   <property name="tickPosition">
                    <enum>QSlider::NoTicks</enum>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item>
                <widget class="QPushButton" name="Sorting_Start_Button">
                 <property name="text">
                  <string>START AUTOMATIC SORTING</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QWidget" name="widget" native="true"/>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="Neural_Network_Tab">
         <attribute name="title">
          <string>Neural Network</string>
         </attribute>
         <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="2,3,6">
          <item>
           <layout class="QVBoxLayout" name="Neural_Network_ButtonLayout" stretch="0,0,0">
            <property name="topMargin">
             <number>0</number>
            </property>
            <property name="bottomMargin">
             <number>0</number>
            </property>
            <item>
             <widget class="QPushButton" name="Start_Neural_Network">
              <property name="text">
               <string>Start Neural Engine</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="Scan_Image">
              <property name="text">
               <string>Scan current image</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="Draw_Preview">
              <property name="text">
               <string>Draw preview</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="Neural_Network_Preview" name="Neural_Networ_Output_Preview" native="true">
            <property name="styleSheet">
             <string notr="true">background: #202020</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="NN_Interface" name="Neural_Network_Output"/>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1088</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MpvWidget</class>
   <extends>QOpenGLWidget</extends>
   <header location="global">mpvwidget.h</header>
  </customwidget>
  <customwidget>
   <class>AvgImgPlot</class>
   <extends>QWidget</extends>
   <header location="global">avgimgplot.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>NN_Interface</class>
   <extends>QTextBrowser</extends>
   <header>nn_interface.h</header>
  </customwidget>
  <customwidget>
   <class>Neural_Network_Preview</class>
   <extends>QWidget</extends>
   <header location="global">neural_network_preview.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
